### ISSUE TITLE
Random generation during GRPO training process

### ISSUE THREAD

**Issue by ShianZheng** [modelscope/ms-swift#5523]
Base model: Qwen2.5-VL-7B-Instruct  after SFT

After about 200 steps, the rollout model collapsed and generated some random char or emoji like 

😉✍✍✨️✍ॐ📝🔥💥💥🔥🍃🌙ựa👧🔥🚀🌈RTOS🔍🔒➡🦑👨👩🔥👩👨💡🏠.AI🚀🌌🦿👩💥🦾📐❤🌌🧠🔍💦🏠.AI📚🌟🦿🏠.AI🌌deque🧩💰💸👧💻🔥🏠.AI🔍RTOS🦴📦👨👩❤🦾👩👦.AI🏠.AI👨👩🌸💥🦵🦠📦🚀📦💦🏠.AI🔥💎🌌🔮🦾💻💪♠️🧱🏠.AI🌐🌐🚫🔍☁🚀 ula🛠️👨👧🏠.AI💪💡🔥-pythondeque👨🐶🛠🏠.AI🤖🔮👨👩🌟🏠.AI📐🦾🦾deque🌈👨💥🦾🚀.AI🏠.AI🧳🧤💎🔍🛠👩👧🌙💡🔍👨🔒❄🚨💥🔥💡🔒👨🔧💡🧱👀🔒💨💥💔❄🔒💔🪶🚪👩🔍.AI🔮🦾 ula🏠.AI🌐🔥💡🌟🏠.AI🏠.AI💪🏠.AI👩🦾-pythondeque🦲👨🔧👩❄🌐🦺❤ ula/apt-cache🔍🔍RTOS🚀🔥💪❄🌌🏠.AI👧👧👦🔒💥🧠🛰🦾🔥✨💥🏠.AI📦👀🔍🌸💨.AI👨👧👨💥💡🌌🌐🌐📐🌸👧📦 kaps ↓🔍-pythondeque🌈🧳💪💡🌌🌙❤️🛠🏠.AI💎👦💻💪💰💥📦🔍☁.AI🌸🏠.ReLU🔥.AI🌟🌈-pythondeque🌱🔍👀👩🌸👧🚀🚀💨🏠.AI👨.AI👁.AI🏠.AI🔮🏠.AI👩🔥🐍🏠.AI👩🏠.AI-pythondeque🌌🔍🏠.AI💻🔧💡💔月薪💰🪖🌈📦月薪����_AF👩.AI👧💡🏠.AI🏠.AI🌈🔧🌐.AI❄ ula🔍🌙🌟🏠.AI📐🚀💔🔒🚨💻🤖.AI🌐🔍🏠.AI💡🔍.AI💪��STM/apt-sync💡.AI🔧👩月薪🩹/repos📦.AI👩_AI👧💥🏠.AI👨👨💡🌌月薪��.AI🪳🧱👀👦🦶"

I ve tried many different param set and all of them did not work. here's my script and some fig
swift rlhf \
    --rlhf_type grpo  \
    --model $prev_rl_model \
    --train_type full \
    --dataset my_dataset \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8008 \
    --vllm_gpu_memory_utilization 0.9 \
    --vllm_tensor_parallel_size 1 \
    --async_generate true \
    --torch_dtype bfloat16 \
    --system system_prompt.txt \
    --num_train_epochs 2 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 1e-6 \
    --save_steps 50 \
    --save_total_limit 10 \
    --logging_steps 5 \
    --output_dir $GRPO_OUTPUT_DIR \
    --gradient_accumulation_steps 16 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --max_completion_length 1024 \
    --external_plugins orm.py \
    --reward_funcs result_match \
    --reward_weights 1 \
    --num_generations 8 \
    --temperature 0.8 \
    --top_p 0.85 \
    --beta 0.01

<img width="1646" height="335" alt="Image" src="https://github.com/user-attachments/assets/a59d0b37-f60a-4c36-9c23-6df64b9603a6" />

**Comment by hjh0119** [comment]
Could you retry with async_generate=false? It might introduce some training instability

**Comment by ShianZheng** [comment]
> Could you retry with async_generate=false? It might introduce some training instability
 working on it.

**Comment by ShianZheng** [comment]
> Could you retry with async_generate=false? It might introduce some training instability

Ok, working on it. btw is it related to the external mode? I m not quite sure about how rollout model is updated in this mode

**Comment by hjh0119** [comment]
https://github.com/modelscope/ms-swift/blob/v3.7.2/swift/trainers/rlhf_trainer/grpo_trainer.py#L546-L607

**Comment by ShianZheng** [comment]
> Could you retry with async_generate=false? It might introduce some training instability

<img width="547" height="651" alt="Image" src="https://github.com/user-attachments/assets/c8e63806-d9b3-4aae-bcd9-8d8621ec8949" />

it still have the same problem. The result reward looks just fine because of the recovery, but the completion length went down and the thinking process is completely lost.

**Comment by hjh0119** [comment]
I believe this is expected behavior. The reward curve looks normal, and in your task the model can receive rewards even without producing any explicit “thinking” content. Additionally, the default loss_type averages at the completion level, which introduces a length bias. You could try adding length-aware rewards and switch to token-level loss with --loss_type bnpo

**Comment by ShianZheng** [comment]
It didn't get a higher reward, so shouldn't it try different actions to find the right answer? Not sure if the length-aware reward helped or just made things harder.
